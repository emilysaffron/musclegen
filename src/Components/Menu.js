import styled from "@emotion/styled";
// This is just a simple mapping function, only used by one component - it doesn't need to be a helper
import GetButtonNames from "../Helpers/GetButtonNames";
const StyledButtons = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

const Menu = ({ labels, use }) => {
  // This is actually returning Button components, rather than just their names (which suggests string values) - consider renaming.
  const buttons = GetButtonNames(labels, use);
  return <StyledButtons>{buttons}</StyledButtons>;
};

export default Menu;
