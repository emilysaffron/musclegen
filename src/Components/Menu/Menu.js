import styled from "@emotion/styled";
import MenuButton from "../MenuButton/MenuButton";

const StyledButtons = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

const Menu = ({ labels, dropdown}) => {
  const GetMenuButtons = (labels, dropdown) => {
    return labels.map((item) => (
      <MenuButton label={item} key={item} dropdown={dropdown} />
    ));
  };
  const buttons = GetMenuButtons(labels, dropdown);
  return <StyledButtons>{buttons}</StyledButtons>;
};

export default Menu;
