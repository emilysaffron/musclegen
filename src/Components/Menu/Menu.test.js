import renderer from "react-test-renderer";
import Menu from "./Menu";

it("Menu Snapshot", () => {
  const tree = renderer.create(<Menu labels={[]} dropdown={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
