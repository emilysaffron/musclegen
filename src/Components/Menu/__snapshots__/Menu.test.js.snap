// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Snapshot 1`] = `
<div
  className="css-deizzc"
/>
`;
